src/crimson/osd/osd_operations/client_request.cc

```diff
@@ -23,10 +23,18 @@ namespace crimson::osd {
 void ClientRequest::Orderer::requeue(Ref<PG> pg)
 {
   LOG_PREFIX(ClientRequest::Orderer::requeue);
-  for (auto &req: list) {
-    DEBUGDPP("requeueing {}", *pg, req);
-    req.reset_instance_handle();
-    std::ignore = req.with_pg_process(pg);
+  std::list<ClientRequest*> to_requeue;
+  for (auto &req : list) {
+    to_requeue.emplace_back(&req);
+  }
+  // Client requests might be destroyed in the following
+  // iteration leading to short lived dangling pointers
+  // to those requests, but this doesn't hurt as we won't
+  // dereference those dangling pointers.
```

Comment:
Can you please explain how the Client requests might be destroyed?
Is it because we re-queue them or unrelated to re-queuing?

Reply:
Just as shown in the log in https://tracker.ceph.com/issues/67068, if a request gets requeued and get discarded, all the process of that request can be within the same seastar continuation, because seastar process immediately resolved futures in the same continuation when running without NDEBUG. In this case, before we run into the next loop, the iterator based on which we find the next item would have been invalidated, and we get undefined behaviours.